docs/docs/1.0.6/documentation/health.md

```diff
@@ -0,0 +1,160 @@
+# Health checks
+
+Health is an extremely important part of the microservice architecture for various reasons. Load balancers rely on the 
+health status of the application while picking endpoint from load balancer set. Users want the application state to be 
+observable through the GUI or CLI. Orchestrators like Kubernetes also want to know the application status to manage 
+lifecycle of the application. 
+
+Kuma supports several aspects of the health checking. There are two policies which allows configuring active and passive:
+[Health Check](../policies/health-check.md) and [Circuit Breaker](../policies/circuit-breaker.md).
+
+Kuma is able to track the status of the Envoy proxy. If grpc stream with Envoy is disconnected then Kuma considers this 
+proxy as offline. 
+
+Also, every `inbound` in the Dataplane model has `health` section:
+
+```yaml
+type: Dataplane
+mesh: default
+name: web-01
+networking:
+  address: 127.0.0.1
+  inbound:
+    - port: 11011
+      servicePort: 11012
+      health:
+        ready: true
+      tags:
+        kuma.io/service: backend
+        kuma.io/protocol: http
+```
+
+This `health.ready` status is intended to show the status of the application itself. It is set differently depending on 
+the environment ([Kubernetes](#kubernetes-probes) or [Universal](#universal-probes)), but it's treated the same way 
+regardless of the environment:
+
+- if proxy status is `Offline`, then Dataplane is `Offline`:
+- if proxy status is `Online`:
+  - if all inbounds are ready then Dataplane is `Online`
+  - if all inbounds are not ready then Dataplane is `Offline`
+  - if at least one of the inbounds is not ready then Dataplane is `Partially degraded` 
+  - if inbound is not ready then it's not included in the load-balancer set which means it doesn't receive the traffic
+  - if all inbounds which implement the same service are ready then service is `Online`
+  - if all inbounds which implement the same service are not ready then service is `Offline`
+  - if at least one of the inbounds which implement the same service is not ready then service is `Partially degraded`
```

Comment:
I feel like you are mixing a lot of things here
* Status of the dataplane
* When dataplane is included in the traffic
* Status of the service

I think what I need to know as a user is
* Dataplane is taken into account in traffic when `health.ready` is set to `true` OR the `health` section is missing. 
* How statuses are computed (Dataplane and Service separately)